airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/operation/SnowflakeStagingClient.kt

```diff
@@ -84,7 +111,8 @@ class SnowflakeStagingClient(private val database: JdbcDatabase) {
             filePath,
             stageName,
             stagingPath,
-            Runtime.getRuntime().availableProcessors()
+            // max allowed param is 99, we don't need so many threads for a single file upload
+            minOf(Runtime.getRuntime().availableProcessors(), 4)
```

Comment:
why not `minOf(..., 99)`? wouldn't this restrict us to 4 threads in most cases?

Reply:
in our cloud case it is always 1, our pod config is 1 cpu i believe based on my previous dev release run.

Reply:
@gisripa 
1. The pod config could also run on 2 cpus by values.yaml definition ain't im wrong? (worker resources)
2. The update will be released on the next snowflake dest version? or the current one?

Reply:
@talnidam surely can use `minOf(..,99)` however there is no added benefit in the way we upload 1 file only. We use a 200M file and only 1 file per PUT and snowflake chunks it if > 64M which I'm guessing won't be more than 3 or 4 chunks.  The advantage of this shows up if we do PUT with a `directory/*` with many files in the directory. Also the default is 4 in snowflake settings if not provided. so just used that as max.